beacon-chain/db/kv/state_summary.go

```diff
@@ -78,3 +89,25 @@ func (s *Store) stateSummaryBytes(ctx context.Context, blockRoot [32]byte) ([]by
 
 	return enc, err
 }
+
+// This saves all cached state summary objects to DB, and clears up the cache.
+func (s *Store) saveCachedStateSummariesDB(ctx context.Context) error {
+	summaries := s.stateSummaryCache.getAll()
+	if err := s.db.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(stateSummaryBucket)
+		for _, s := range summaries {
+			enc, err := encode(ctx, s)
```

Comment:
Ideally this sort of operation should be done outside of the bolt tx to save on time the db lock is held, but no big deal, just something to keep in mind if this method is slow in the future

Reply:
Good point. Moved encoding outside of the tx:
```go
	encs := make([][]byte, len(summaries))
	for i, s := range summaries {
		enc, err := encode(ctx, s)
		encs[i] = enc
	}
	if err := s.db.Update(func(tx *bolt.Tx) error {
		bucket := tx.Bucket(stateSummaryBucket)
		for i, s := range summaries {
			bucket.Put(s.Root, encs[i])
		}
	});
```